### PR TITLE
feat: Tab list pill variant + fullWidth

### PR DESCRIPTION
**Pull request by nicolls1**
This PR adds a `pill` variant to tabs. It also adds a `fullWidth` prop to both `default` and `pill` variants. The `pill` variant styling is intended to match the styling of `<RadioGroupMinimal`. Also note, `default` tabs have text that wraps where as `pill` tabs text truncates with overflow ellipsis.

### AFTER

Screen Shot 2021-11-16 at 9 17 17 AM

Screen Shot 2021-11-16 at 9 18 59 AM

### CODE REVIEW

File: src/components/Tabs/TabList.tsx [hunk 1]
```diff
@@ -27,18 +34,20 @@ export interface TabListProps {
   className?: string;
 }
 
-export const TabList: React.FC<TabListProps> = ({
+export const TabList: React.FC<TabListProps & Partial<TabListOptions>> = ({
```
Comment (nicolls1) [inline src/components/Tabs/TabList.tsx:37]: Thought passing the types of the base component for these would be nice. Any concerns with this?

Comment (ynotdraw) [reply src/components/Tabs/TabList.tsx:37]: I _think_ it'd be okay, my only worry may be there could be something we don't want to support from a design perspective (e.g., `orientation`, etc.).

We had initially only exposed the props we approved from a design perspective intentionally ( cc @dexterca ) .  I'm a bit torn personally on exposing this, my gut feeling says leave things as-is for now until we have a case where we specifically want one of these additional props, but I don't have strong opinions either way 😄

File: src/components/Tabs/TabsContext.tsx [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
 
-export const TabsContext = React.createContext<any>({ selectedId: null });
+export interface TabsState {
+  tabState: TabStateReturn;
+  variant: 'default' | 'pill';
+}
+
+export const TabsContext = React.createContext<TabsState>({
+  // only will happen when tabs components are used outside of the context
+  tabState: {} as any,
```
Comment (ynotdraw) [inline src/components/Tabs/TabsContext.tsx:11]: Nice, yeah thanks for cleaning this up.  I completely missed this when removing `selectedId` before 🙇

File: src/components/Tabs/TabList.tsx (hunk 1 above)
Comment (nicolls1) [reply src/components/Tabs/TabList.tsx:37]: Sounds good, lets see if there are any other opinions else we can hold off

**Comment by nicolls1** [comment]
Waiting on actual pill styles to merge

**Comment by dexterca** [comment]
👀 I'm going to take a look at pill styles today. I plan on just adding my changes to this PR unless someone objects to that.

**Comment by nicolls1** [comment]
Checked it out locally and LGTM!

File: src/components/Tabs/Tab.tsx [hunk 3]
```diff
@@ -48,6 +48,61 @@ export const useStyles = makeStyles(
         opacity: 0.4,
       },
     },
+    pill: {
+      borderBottom: 'none',
+      borderRadius: '50px',
```
Comment (ynotdraw) [inline src/components/Tabs/Tab.tsx:53]: Should this use `theme.pxToRem(50)` instead to be consistent with the `borderRadius` on line 65 and the `borderRadius` set here https://github.com/lifeomic/chroma-react/pull/227/files#diff-0243aba83c9530a47511fd3f1ac4a05bc5dbfafa61918171d15d0f7a6c456994R21?

File: src/components/Tabs/Tabs.test.tsx [hunk 4]
```diff
@@ -21,6 +21,38 @@ test('it renders Tabs', async () => {
...
+        <Tab stopId="tab2">Tab 2</Tab>
+      </TabList>
+    </Tabs>
+  );
+
+  const root = await findByTestId(testId);
+  const options = root.querySelectorAll('[role="tab"]');
+  expect(options).toBeTruthy();
+  expect(options.length).toEqual(2);
+});
```
Comment (ynotdraw) [inline src/components/Tabs/Tabs.test.tsx:54]: Could this test verify a `className` has `fullWidth` somewhere via `toHaveClass()`?

File: src/components/Tabs/Tabs.test.tsx [hunk 5]
```diff
@@ -21,6 +21,38 @@ test('it renders Tabs', async () => {
...
```
Comment (ynotdraw) [inline src/components/Tabs/Tabs.test.tsx:38]: Is there a way to check that the `pill` variant is applied somewhere in the DOM?  May be able to leverage `toHaveClass()` and assert on that.

File: stories/components/Tabs/default.md [hunk 6]
```diff
@@ -37,6 +37,15 @@ A class name can be provided.
...
 
+### Full Width
+
+To set the width of tabs to take 100% of the width, use the `fullWidth`
+prop.
+
+```jsx
+<Tabs fullWidth>…</Tabs>
+```
+
```
Comment (ynotdraw) [inline stories/components/Tabs/default.md:48]: Should the `variant` prop be added to the docs as well for visibility?

File: src/components/Tabs/Tab.tsx (hunk 3 above)
Comment (dexterca) [reply src/components/Tabs/Tab.tsx]: ✅

File: stories/components/Tabs/default.md (hunk 6 above)
Comment (dexterca) [reply stories/components/Tabs/default.md:58]: ✅

File: src/components/Tabs/Tabs.test.tsx (hunk 4 above)
Comment (dexterca) [reply src/components/Tabs/Tabs.test.tsx:54]: @nicolls1 can you take a look at updating per @ynotdraw's suggestion?

File: src/components/Tabs/Tabs.test.tsx (hunk 5 above)
Comment (dexterca) [reply src/components/Tabs/Tabs.test.tsx:38]: @nicolls1 can you take a look at updating per @ynotdraw's suggestion?

Comment (nicolls1) [reply src/components/Tabs/Tabs.test.tsx:40]: Pushed

File: src/components/Tabs/Tabs.test.tsx (hunk 4 above)
Comment (nicolls1) [reply src/components/Tabs/Tabs.test.tsx]: Pushed

**Review by ynotdraw — APPROVED** [review]

**Review by smolster — APPROVED** [review]
Nice work!

**Pull request merged**